### DISCUSSION TITLE
Passing a variable to middleware

### DISCUSSION THREAD

**Discussion by dmeinke-etc** [adonisjs/core#3907] [Help]
I want to ensure a user has access to the client page they are attempting to reach with the
 `Route.get('/client/:id', 'ClientController.index').middleware('clientAccess')`
But I want to make sure the :id is something the user has access too.
I have an array of ID's in a session called user_access. In the middleware handle function I want to test if it is true  

```
if(session.get('user_access').includes(:id)){
return true
}
```

But how do I pass the :id to the middleware?

**Comment by WeyIin** [comment]
Hi @dmeinke-etc ,

Check this out, it is probably exactly what you need:
https://docs.adonisjs.com/guides/authorization

e.g.

```typescript
import Route from '@ioc:Adonis/Core/Route'
import Post from 'App/Models/Client'

Route.get('client/:id', async ({ bouncer, request }) => {
  const client= await Client.findOrFail(request.param('id'))

  await bouncer.authorize('viewClient', client)
})
```
